src/main/java/xfacthd/oretexgen/client/loader/OreMetadata.java

```diff
@@ -1,46 +1,82 @@
 package xfacthd.oretexgen.client.loader;
 
+import com.google.gson.JsonElement;
 import com.google.gson.JsonObject;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.server.packs.metadata.MetadataSectionSerializer;
 import net.minecraft.server.packs.resources.Resource;
+import org.jetbrains.annotations.Nullable;
 import xfacthd.oretexgen.OreTextureGenerator;
 
 import java.io.IOException;
 
 public final class OreMetadata
 {
     private static final String DEFAULT_BACKGROUND = "minecraft:block/stone";
-    private static final OreMetadata DEFAULT = new OreMetadata(DEFAULT_BACKGROUND, false);
+    private static final OreMetadata DEFAULT = new OreMetadata(DEFAULT_BACKGROUND, ShadowMetadata.fromJson(new JsonObject()));
     private static final Serializer SERIALIZER = new Serializer();
 
     private final ResourceLocation background;
-    private final boolean generateShadow;
 
-    private OreMetadata(String background, boolean generateShadow)
+    @Nullable
+    private final ShadowMetadata shadowMetadata;
+
+    private OreMetadata(String background, @Nullable ShadowMetadata shadowMetadata)
     {
         this.background = new ResourceLocation(background);
-        this.generateShadow = generateShadow;
+        this.shadowMetadata = shadowMetadata;
     }
 
     public ResourceLocation getBackground()
     {
         return background;
     }
 
-    public boolean shouldGenerateShadow()
+    @Nullable
+    public ShadowMetadata getShadowMetadata()
     {
-        return generateShadow;
+        return shadowMetadata;
     }
 
-
-
```

Comment:
Please keep these lines, I prefer adding some space between non-static methods and static helpers below them.